Examples/Io/Root/include/ActsExamples/Io/Root/RootDigitizationWriter.hpp

```diff
@@ -0,0 +1,238 @@
+      std::string treeName = "vol" + std::to_string(vID);
+      if (lID > 0) {
+        treeName += "_lay" + std::to_string(lID);
+      }
+      if (mID > 0) {
+        treeName += "_mod" + std::to_string(mID);
+      }
+      setupTree(treeName);
+    }
+
+    /// Setup the dimension depended branches
+    ///
+    /// @tparam parset_t Type of the parameter set
+    ///
+    /// @param ps A dummy bound parameter set
+    template <typename parset_t>
+    void setupBoundRecBranches(const parset_t& ps) {
+      if (ps.template contains<Acts::eBoundLoc0>()) {
+        tree->Branch(std::string("rec_" + bNames[Acts::eBoundLoc0]).c_str(),
+                     &recBound[Acts::eBoundLoc0]);
+      }
+      if (ps.template contains<Acts::eBoundLoc1>()) {
+        tree->Branch(std::string("rec_" + bNames[Acts::eBoundLoc1]).c_str(),
+                     &recBound[Acts::eBoundLoc1]);
+      }
+      if (ps.template contains<Acts::eBoundPhi>()) {
+        tree->Branch(std::string("rec_" + bNames[Acts::eBoundPhi]).c_str(),
+                     &recBound[Acts::eBoundPhi]);
+      }
+      if (ps.template contains<Acts::eBoundTheta>()) {
+        tree->Branch(std::string("rec_" + bNames[Acts::eBoundTheta]).c_str(),
+                     &recBound[Acts::eBoundTheta]);
+      }
+      if (ps.template contains<Acts::eBoundTime>()) {
+        tree->Branch(std::string("rec_" + bNames[Acts::eBoundTime]).c_str(),
+                     &recBound[Acts::eBoundTime]);
+      }
+    }
+
+    /// Convenience function to register idenfication
+    ///
+    /// @param eventNr The event number
+    /// @param geoID The geometry identifier of the measurement
+    void fillIdentification(int evnt, Acts::GeometryIdentifier geoId) {
+      eventNr = evnt;
+      volumeID = geoId.volume();
+      layerID = geoId.layer();
+      surfaceID = geoId.sensitive();
+    }
+
+    /// Convenience function to register the truth parameters
+    ///
+    /// @param lp The true local position
+    /// @param xt The true 4D global position
+    /// @param dir The true particle direction
+    void fillTruthParameters(const Acts::Vector2D& lp, const Acts::Vector4D& xt,
+                             const Acts::Vector3D& dir) {
+      trueBound[Acts::eBoundLoc0] = lp[Acts::eBoundLoc0];
+      trueBound[Acts::eBoundLoc1] = lp[Acts::eBoundLoc1];
+      trueBound[Acts::eBoundPhi] = Acts::VectorHelpers::phi(dir);
+      trueBound[Acts::eBoundTheta] = Acts::VectorHelpers::theta(dir);
+      trueBound[Acts::eBoundTime] = xt[Acts::eTime];
+
+      trueGx = xt[Acts::ePos0];
+      trueGy = xt[Acts::ePos1];
+      trueGz = xt[Acts::ePos2];
+    }
+
+    /// Convenience function to fill bound parameters
+    ///
+    /// @tparam measurement_t Type of the parameter set
+    ///
+    /// @param m The measurement set
+    template <typename measurement_t>
+    void fillBoundMeasurement(const measurement_t& m) {
+      auto fullVect = m.projector().transpose() * m.parameters();
```

Comment:
Maybe I'm wrong, but could we really use the projector transpose to project the measurement parameter set to the full bound parameter set? I thought the projector could only do something the other way around.

Reply:
Shouldn't it just expand the parameters to a full set (and fill with 0s), that's what I want here ... - and it seems to work.

Reply:
It does work since our projectors only contain ones or zeros. I am not sure if it would work with more general projection matrices.

Reply:
Have you checked if the values of phi and theta are correct? If yes, then forget what I said...

Reply:
Have checked with the time parameter (as shown above).

Reply:
I think it's working because the time parameter is contained in the measurement parset.  If the measurement par indices does not contain the phi and theta for example, the projected phi and theta will be just zero.

Reply:
But it does not hurt to include them though they could be zero.

Reply:
Sorry, my bad. This `fillBoundMeasurement` is just filling the values, not adding the branches in the tree. So should be fine.